mozilla/contain-facebook
# Add Custom Sites to the Facebook Container

This PR adds buttons to the doorhanger panel for users to add/remove the active page/domain to the Facebook container. 

Fixes #422 
Fixes #550  

**PR To-do List**
- [x] Update onboarding panel copy/images
- [x] Add tooltip text
- [x] Fix issue where "Add this site" shows up on default FB URLs (from [FACEBOOK_DOMAINS](https://github.com/mozilla/contain-facebook/blob/master/src/background.js#L7-L23)) 
- [x] Add test directions to this PR 

## Testing Steps

Use the following directions to verify this PR works as is. 

1. **Run the add-on.** _(This will open up the add-on in a temporary new browser)_

```
$ web-ext run -s src/ 
```

2. **Go to the following sites:** 

- Non-Facebook tracking site: 
  - https://duckduckgo.com/ (There should be no Facebook trackers blocked on this site) 
- Facebook domain 
  - https://facebook.com/
-  Facebook tracking site
  - https://maxxcrawford.github.io/social-implementation-examples/facebook/login/
  - https://buzzfeed.com/

3. Once on the site, verify that trackers are detected as expected. 

4. Click on the Facebook Container add-on icon within the browser chrome to trigger the panel:   ![image](https://user-images.githubusercontent.com/2692333/71749332-bce00400-2e3a-11ea-9019-4540b8068d19.png)

5. Attempt to add that site to the Facebook Container: 

- Non-Facebook tracking site and Facebook tracking site should be able be to added by clicking the following button:
![image](https://user-images.githubusercontent.com/2692333/71749368-d4b78800-2e3a-11ea-8c6d-501ed8258413.png)

Once this button is clicked, the expected action is for the tab to close and reopen the same URL **INSIDE** the Facebook Container. 

6. Once added, attempt to remove the newly added site from the Facebook Container: 
- Confirm you're in the Facebook Container
![image](https://user-images.githubusercontent.com/2692333/71749446-13e5d900-2e3b-11ea-90f1-5412334c20af.png)

- Click on the Facebook Container add-on icon within the browser chrome to trigger the panel: 
![image](https://user-images.githubusercontent.com/2692333/71749561-66bf9080-2e3b-11ea-869c-7421b09c1109.png)

  - On default Facebook Container sites (like Facebook.com and Instagram.com), these sites cannot be removed. 
  - On non-default sites, it should close and reload the page **OUTSIDE** of the Facebook Container. 



## Screenshots

No trackers detected panel (outside of FB Container): 
![image](https://user-images.githubusercontent.com/2692333/71698512-71b5ea80-2d81-11ea-8c6c-1a11c3992247.png)


Trackers detected panel (outside of FB Container): 
![image](https://user-images.githubusercontent.com/2692333/71698562-91e5a980-2d81-11ea-881d-49bc34cb0e0b.png)


Custom site added to FB Container
![image](https://user-images.githubusercontent.com/2692333/71698553-8befc880-2d81-11ea-8e77-9d0d239f7b6e.png)


On default Facebook domain: 
![image](https://user-images.githubusercontent.com/2692333/71698536-86927e00-2d81-11ea-988e-f8596693316a.png)


On default Facebook domain (hover tooltip): 
![image](https://user-images.githubusercontent.com/2692333/71698521-82666080-2d81-11ea-9689-7103a49659b6.png)


In-page UI Panel (Revised Copy + Image): 
![image](https://user-images.githubusercontent.com/2692333/71698397-03712800-2d81-11ea-889d-5697d06156c7.png)

New Panel for Adding Custom Site: 
![image](https://user-images.githubusercontent.com/2692333/71698411-0c61f980-2d81-11ea-9d58-c0a0dd7eb816.png)
